src/attr/_make.py

```diff
@@ -680,6 +684,8 @@ def fmt_setter_with_converter(attr_name, value_var):
                                                     a.name))
         names_for_globals[val_name] = a.validator
         names_for_globals[attr_name] = a
+    if post_init:
+        lines.append("self.__post_init__()")
```

Comment:
I'm voting post validation/conversion. The name `__post_init__` itself suggests it's called after the generated `__init__`, so I'd expect fields to be validated, converted and set. Having it called post-conversion lowers the amount of possible states we can expect the object to be in.